include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/GenerateRobolectricPreviewTestsExtension.kt

```diff
@@ -0,0 +1,236 @@
+          "$libraryName version",
+          "UTF-8"
+        ) + "\n" +
+          "testImplementation(\"$libraryName:version\")"
+      )
+    }
+  }
+
+  val requiredLibraries = listOf(
+    "io.github.takahirom.roborazzi:roborazzi-compose",
+    "io.github.takahirom.roborazzi:roborazzi",
+    "junit:junit",
+    "org.robolectric:robolectric",
+    "com.github.sergio-sastre.ComposablePreviewScanner:android",
+  )
+  requiredLibraries.forEach { checkExists(it) }
+}
+
+abstract class GeneratePreviewScreenshotTestsTask : DefaultTask() {
+  @get:OutputDirectory
+  abstract val outputDir: DirectoryProperty
+
+  @get:Input
+  var scanPackageTrees: ListProperty<String> = project.objects.listProperty(String::class.java)
+
+  @TaskAction
+  fun generateTests() {
+    val testDir = outputDir.get().asFile
+    testDir.mkdirs()
+
+    val packagesExpr = scanPackageTrees.get().joinToString(", ") { "\"$it\"" }
+    val scanPackageTreeExpr = ".scanPackageTrees($packagesExpr)"
+
+    val className = "RoborazziPreviewParameterizedTests"
+    File(testDir, "$className.kt").writeText(
+      """
+            import org.junit.Test
+            import org.junit.runner.RunWith
+            import org.robolectric.ParameterizedRobolectricTestRunner
+            import org.robolectric.annotation.Config
+            import org.robolectric.annotation.GraphicsMode
+            import com.github.takahirom.roborazzi.*
+            import sergio.sastre.composable.preview.scanner.core.preview.ComposablePreview
+            import sergio.sastre.composable.preview.scanner.android.AndroidPreviewInfo
+            import sergio.sastre.composable.preview.scanner.android.AndroidComposablePreviewScanner
+            import sergio.sastre.composable.preview.scanner.android.screenshotid.AndroidPreviewScreenshotIdBuilder
+
+
+            @RunWith(ParameterizedRobolectricTestRunner::class)
+            @GraphicsMode(GraphicsMode.Mode.NATIVE)
+            class $className(
+                private val preview: ComposablePreview<AndroidPreviewInfo>,
+            ) {
+
+                companion object {
+                    val previews: List<ComposablePreview<AndroidPreviewInfo>> by lazy {
+                        AndroidComposablePreviewScanner()
+                            $scanPackageTreeExpr
+                            .getPreviews()
+                    }
+                    @JvmStatic
+                    @ParameterizedRobolectricTestRunner.Parameters
+                    fun values(): List<ComposablePreview<AndroidPreviewInfo>> =
+                        previews
+                }
+                
+                fun createScreenshotIdFor(preview: ComposablePreview<AndroidPreviewInfo>) = 
+                  AndroidPreviewScreenshotIdBuilder(preview)
+                         .ignoreClassName()
+                         .build()
+                
+                @GraphicsMode(GraphicsMode.Mode.NATIVE)
+                @Config(sdk = [30])
```

Comment:
So my understanding is, that for the first version, we do not handle the preview properties right?
I mean, locale, uiMode, fontscale, etc... are easy to support:
https://github.com/sergio-sastre/Android-screenshot-testing-playground/blob/master/lazycolumnscreen-previews/roborazzi/src/test/java/snapshot/testing/lazycolumn_previews/roborazzi/utils/RobolectricPreviewInfosApplier.kt

Other more complicated, like apiLevel, device etc. can be supported in next versions.

What do you think?

in that case, I'd also document which preview parameters are supported

Reply:
@sergio-sastre @yschimke
Thank you for all your reviews!
I've created a new PR to handle the preview settings. Could you please take a look
https://github.com/takahirom/roborazzi/pull/427/files